native_client/ctcdecode/path_trie.cpp

```diff
@@ -39,7 +39,8 @@ PathTrie* PathTrie::get_path_trie(int new_char, int new_timestep, float cur_log_
   auto child = children_.begin();
   for (child = children_.begin(); child != children_.end(); ++child) {
     if (child->first == new_char) {
-      if (child->second->log_prob_c < cur_log_prob_c) {
+      if (child->second->log_prob_c < cur_log_prob_c &&
+          child->second->children_.size() == 0) {
```

Comment:
Maybe worth commenting that if we update elements that are not leafs, we will overwrite earlier timecodes.